ple/games/donkeykong/Board.py

```diff
@@ -0,0 +1,359 @@
+                    self.map[i * 5 + k][brokenLadderPos] = self.map[i * 5 + k][brokenLadderPos + 1] = 2
+                for k in range(3 + brokenRand, 5):
+                    self.map[i * 5 + k][brokenLadderPos] = 2
+                for k in range(3 + brokenRand2, 5):
+                    self.map[i * 5 + k][brokenLadderPos + 1] = 2
+
+    # Create the holes on each floor (extreme right and extreme left)
+    def makeHoles(self):
+        for i in range(3, (self.__height / (15 * 5) - 1)):
+            for k in range(1, 6):  # Ladders wont interfere since they leave 10 blocks on either side
+                if i % 2 == 0:
+                    self.map[i * 5][k] = 0
+                else:
+                    self.map[i * 5][self.__width / 15 - 1 - k] = 0
+
+    '''
+    This is called once you have finished making holes, ladders, walls etc
+    You use the 2D map to add instances to the groups
+    '''
+
+    def populateMap(self):
+        for x in range(len(self.map)):
+            for y in range(len(self.map[x])):
+                if self.map[x][y] == 1:
+                    # Add a wall at that position
+                    self.Walls.append(OnBoard(pygame.image.load('Assets/wood_block.png'), (y * 15 + 15 / 2, x * 15 + 15 / 2)))
+                elif self.map[x][y] == 2:
+                    # Add a ladder at that position
+                    self.Ladders.append(OnBoard(pygame.image.load('Assets/ladder.png'), (y * 15 + 15 / 2, x * 15 + 15 / 2)))
+
+    # Check if the player is on a ladder or not
+    def ladderCheck(self, laddersCollidedBelow, wallsCollidedBelow, wallsCollidedAbove):
+        if laddersCollidedBelow and len(wallsCollidedBelow) == 0:
+            for ladder in laddersCollidedBelow:
+                if ladder.getPosition()[1] >= self.Players[0].getPosition()[1]:
+                    self.Players[0].onLadder = 1
+                    self.Players[0].isJumping = 0
+                    # Move the player down if he collides a wall above
+                    if wallsCollidedAbove:
+                        #print "HIT HEAD"
+                        self.Players[0].updateY(3)
+                    #print "ON LADDER!"
+        else:
+            self.Players[0].onLadder = 0
+
+    # Update all the fireball positions and check for collisions with player
+    def fireballCheck(self):
+        for fireball in self.fireballGroup:
+            fireball.continuousUpdate(self.wallGroup, self.ladderGroup)
+            if fireball.checkCollision(self.playerGroup, "V"):
+                if len(self.Hearts) >= 2:  # Reduce the player's life by 1
+                    self.Fireballs.remove(fireball)
+                    self.Hearts.pop(len(self.Hearts) - 1)
+                    self.Players[0].setPosition((50, 440))
+                    mixer.music.load('Assets/death.wav')
+                    mixer.music.set_volume(1)
+                    pygame.mixer.music.play()
+                    print "YOU DIED"
+                    self.score -= 25
```

Comment:
it would be great if you can return the scores to the main loop to be compounded there. Currently I dont think the main game is aware of when it gains points or losses lives. Maybe reward back the reward gained/lost and lives gained/lost?